client/structs/allochook.go

```diff
@@ -0,0 +1,29 @@
+package structs
+
+import (
+	"sync"
+
+	"github.com/hashicorp/nomad/client/pluginmanager/csimanager"
+)
+
+// AllocHookResources contains data that is provided by AllocRunner Hooks for
```

Comment:
I know you're considering this approach "janky" but I think of this being limited to data coming from pre-run hooks as a feature not a bug. Having messages passed outside of the initial setup of the hooks sounds like it'd be a headache to reason about.

Reply:
Tbh I mostly think it's janky on the receiver side - We have no real signalling that the state is ready, so we rely on the call order implicitly.

I think it's ~fine as long as we never have to rely on async state in a TR hook.